lib/Models/Catalog/Esri/ArcGisMapServerCatalogGroup.ts

```diff
@@ -220,7 +233,14 @@ export class MapServerStratum extends LoadableStratum(
         ArcGisMapServerCatalogItem,
         layerId
       );
+
       if (existingModel === undefined) {
+        const parentLayers = this.findParentLayers(layer.id);
+
+        if (parentLayers.length > 0) {
+          layerId = id + "/" + parentLayers.reverse().join("/");
+        }
+
```

Comment:
I think it would be better to put your new `layerId` at the top of `createMemberFromLayer`. At the the moment your new correct `layerId` is only being used for new items - but the old incorrect id is being used for existing items, new groups or existing groups.

I tested another MapServer group with 4 nested layers, and it is missing the layers that are in nested sub groups 

- https://plan-gis.mapshare.vic.gov.au/arcgis/rest/services/Planning/VicPlan_PlanningSchemeHistory/MapServer

Reply:
Thanks Nick, moving the check for parents to the top of the function means it does correctly pick up the nested groups